src/databricks/labs/dqx/profiler/profiler.py

```diff
@@ -181,6 +181,66 @@ def profile_tables_for_patterns(
             max_parallelism=max_parallelism,
         )
 
+    @telemetry_logger("profiler", "detect_primary_keys_with_llm")
+    def detect_primary_keys_with_llm(
```

Comment:
The generator primary purpose is to generate `is_unique` rules, not to detect primary keys.